components/TwoUp/index.ts

```diff
@@ -0,0 +1,94 @@
+import './styles.css';
+import { PointerTracker, Pointer } from '../../utils/trackDragging';
+
+/**
+ * A split view that the user can adjust. The first child becomes
+ * the left-hand side, and the second child becomes the right-hand side.
+ */
+export default class TwoUp extends HTMLElement {
+  private readonly _handle = document.createElement('div');
+  /**
+   * The position of the split.
+   */
+  private _position = 0;
+  /**
+   * The value of _position when the pointer went down.
+   */
+  private _positionOnPointerStart = 0;
+  /**
+   * Has connectedCallback been called yet?
+   */
+  private _everConnected = false;
+
+  constructor () {
+    super();
+    this._handle.className = 'two-up-handle';
+
+    // Watch for children changes.
+    // Note this won't fire for initial contents,
+    // so _childrenChange is also called in connectedCallback.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
+
+    // Watch for pointers on the handle.
+    const pointerTracker: PointerTracker = new PointerTracker(this._handle, {
+      start: (pointer, event) => {
+        // We only want to track 1 pointer.
```

Comment:
“We only want to track 1 pointer.” but then it early-exits when there is only pointer.

Reply:
That's because `currentPointers` doesn't include this pointer, as we haven't accepted it yet. Maybe that's confusing?

Reply:
A bit yeah. But it might be that I have’t fully grasped this class yet.